app/Nova/User.php

```diff
@@ -809,4 +813,36 @@ private static function adminCanRun(NovaRequest $request): bool
     {
         return $request->user()->hasRole('admin');
     }
+
+    /**
+     * Helper method: Graduation semester is a 6-digit code by default.
+     * Prepares semester data for display, returning empty if no string/invalid
+     * string is entered.
+     */
+    private static function parseGradSemester(?string $sem): string
+    {
+        if ($sem === null) {
+            return '';
+        }
+        if (preg_match('/^[0-9]{4}0[258]$/', $sem) === 0 || preg_match('/^[0-9]{4}0[258]$/', $sem) === false) {
+            return '';
+        }
+        $semcode = substr($sem, 4);
+        $output = '';
+        switch ($semcode) {
+            case '08':
+                $output .= 'Fall ';
+                break;
+            case '02':
+                $output .= 'Spring ';
+                break;
+            case '05':
+                $output .= 'Summer ';
+                break;
+            default:
+                return '';
+        }
+
+        return $output.substr($sem, 0, 4);
```

Comment:
Won't this cut out the season that was just concatenated onto output? it may be easier to read this if we ended by concatenating the `$year` and `$season`.